projects/vdk-plugins/vdk-local-differential-privacy/tests/test_random_response_ingestion.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import os
+from unittest import mock
+
+import numpy as np
+from click.testing import Result
+from vdk.plugin.local_differential_privacy import differential_privacy_plugin
+from vdk.plugin.test_utils.util_funcs import cli_assert_equal
+from vdk.plugin.test_utils.util_funcs import CliEntryBasedTestRunner
+from vdk.plugin.test_utils.util_funcs import jobs_path_from_caller_directory
+from vdk.plugin.test_utils.util_plugins import IngestIntoMemoryPlugin
+
+
+def ingest_data():
+    # CliEntryBasedTestRunner (provided by vdk-test-utils) gives a way to simulate vdk command
+    # and mock large parts of it - e.g passed our own plugins
+    destination_plugin = IngestIntoMemoryPlugin()
+    runner = CliEntryBasedTestRunner(differential_privacy_plugin, destination_plugin)
+
+    result: Result = runner.invoke(
+        ["run", jobs_path_from_caller_directory("random-sampling")]
+    )
+    cli_assert_equal(0, result)
+
+    payload = destination_plugin.payloads[0].payload
+    count_of_true = np.count_nonzero([a["sensitive_key"] for a in payload])
+    count_of_false = len(payload) - count_of_true
+
+    # assert that the plugins have no unintended impact on other columns.
+    assert list(payload[0].keys()) == ["str_key", "sensitive_key"]
+    assert payload[0]["str_key"] == "str"
+
+    # Within the test data the actual of true vs false is 0 to 60.
+    # within the noisy data we want to test to ensure that false is still significantly higher count.
```

Comment:
nit: what about noice shift balance to the left or right? For example, a person fetches a data subset, or a person does select a partial previous data batch then appends this batch?